utils/python/wait_for_tests.py

```diff
@@ -307,19 +307,21 @@ def parse_test_status(file_contents):
     rv = OrderedDict()
     test_name = None
     for line in file_contents.splitlines():
-        if (line.strip().startswith(COMMENT_STATUS)):
-            pass # skip comments
-        elif (len(line.split()) == 3):
-            status, curr_test_name, phase = line.split()
-            if (test_name is None):
-                test_name = curr_test_name
-            if (phase in rv):
-                # Phase names don't matter here, just need something unique
-                rv[phase] = reduce_stati({"%s_" % phase : status, phase : rv[phase]})
-            else:
-                rv[phase] = status
+        tokens = line.strip().split()
+        if (len(tokens)==0 or tokens[0] == COMMENT_STATUS):
+            pass # skip comments and blank lines
         else:
-            logging.warning("In TestStatus file for test '%s', line '%s' not in expected format" % (test_name, line))
+            if len(tokens) >= 3:
+                status, curr_test_name, phase = tokens[:3]
```

Comment:
Due to the number of warnings I see, I'm still a little worried this change will cause this function to misinterpret lines that are not valid status lines as status lines. The prior condition that the number of tokens must exactly equal three was pretty weak, but now it's even weaker. Is there any way to look at content of these three tokens to see if they are sane? For example, status should be a member of the set of valid statuses, curr_test_name should match test_name, and phase should be valid.